### PR TITLE
[Translations] Add config to register translations domains as valid

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #

## Additional info  
Register Translations Domain to make them valid:
```yaml
pimcore:
  translations:
    domains:
      - site_1
      - site_2
```

### CODE REVIEW

File: bundles/CoreBundle/config/pimcore/default.yaml [hunk 1]
```diff
@@ -243,7 +243,13 @@ pimcore:
         profiler:
             routes:
                 - { path: ^/_(profiler|wdt)(/.*)?$ }
-
+        plugin:
+            routes:
+                - { path: ^/plugin(/.*)?$ }
```
Comment (kingjia90) [inline bundles/CoreBundle/config/pimcore/default.yaml]:
```suggestion
```
see https://github.com/pimcore/pimcore/pull/13898

File: models/Translation.php [hunk 2]
```diff
@@ -275,6 +275,17 @@ public static function getByKey(string $id, string $domain = self::DOMAIN_DEFAUL
         return $translation;
     }
 
+    /**
+     *
+     * @return array
```
Comment (blankse) [inline models/Translation.php]:
```suggestion
     * @return string[]
```

File: models/Translation/Dao.php [hunk 3]
```diff
@@ -168,6 +168,11 @@ public function getAvailableDomains(): array
     public function isAValidDomain(string $domain): bool
     {
         try {
+            $translationDomains = $this->model->getRegisteredDomains();
+            if (!in_arrayi($domain, $translationDomains)) {
```
Comment (kingjia90) [inline models/Translation/Dao.php:172]:
```suggestion
            if (!in_array($domain, $translationDomains)) {
```

File: models/Translation/Dao.php [hunk 4]
```diff
@@ -168,6 +168,11 @@ public function getAvailableDomains(): array
...
+                return false;
+            }
+
             $this->db->fetchOne(sprintf('SELECT * FROM translations_%s LIMIT 1;', $domain));
```
Comment (kingjia90) [inline models/Translation/Dao.php:176]: What about adding this?
```suggestion
            $this->db->fetchAssociative(sprintf('SELECT %s, %s, %s, %s FROM translations_%s LIMIT 1',
                $this->db->quoteIdentifier('key'),
                $this->db->quoteIdentifier('language'),
                $this->db->quoteIdentifier('text'),
                $this->db->quoteIdentifier('type'),
                $domain
            ));
```
Similarly to https://github.com/pimcore/pimcore/pull/14087/files, this checks if the table structure is fine to be considered a valid translation domain.

Comment (dvesh3) [reply models/Translation/Dao.php:176]: this is the main reason for adding the config so shouldn't be necessary to check structure.

Comment (kingjia90) [reply models/Translation/Dao.php:176]: ah right, so it's up to the devs to make sure it's well configured

File: models/Translation/Dao.php (hunk 3 above)
Comment (dvesh3) [inline models/Translation/Dao.php]:
```suggestion
            if (!in_array($domain, $translationDomains)) {
```
@kingjia90 I think this suggestion was valid as you table names are case sensitive. why removed it?

Comment (kingjia90) [reply models/Translation/Dao.php]: ah, i wasn't sure about that, i would expect devs to stick with snakecase but yes, it should be case sensitive after all

**Pull request merged**